### PR TITLE
Adds Nars-Ian and clock hawks

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: Ian has recently communed with unspeakable horrors and may now be warped by their power if Nar-Sie passes near them.
add: In order to fight back against their ancient foe Ian, Poly has struck a bargain with Ratvar and will be transformed into a machine by their presence.
/:cl:

A being as powerful as Ian shouldn't be stumped by a mere eldritch horror, so I gave him the ability to accept its gift. Ian is now warped into Nars-Ian on `narsie_act()`. Nars-Ian is considered a cult creature for AI purposes, will not be gibbed by Nar-Sie, and has a spooky sprite to match.

For every pet adjacent to it, Nars-Ian has a 5% chance per `Life()` tick to devour the pet, restoring Nar-Sian's health to full and gibbing them like the trifling weakling they are.

Poly is also transformed into a clock hawk on `ratvar_act()`. Clock hawks have no special effects, but appear mechanical in nature, and count as a clockwork creature.

### CODE REVIEW

**Comment by Poojawa** [comment]
POLY NEEDS THIS ALSO.

With associated yelling on cultist hiveminds.

**Comment by Xhuis** [comment]
Poly is weak, unfit for the Master's gift.
Ian is powerful, strong. Eternal.

**Comment by Poojawa** [comment]
Poly is best pet. His comedic timing surpasses all.


.... CWC Poly maybe? Mechatronic horror.

**Comment by jughu** [comment]
Clockwork poly CHECK THE ARK YOU CHUCKLECUCKS!

**Comment by Poojawa** [comment]
Squawkwork Poly

**Comment by calzilla1** [comment]
If were making pets canonically syndicate personnel then make runtime teleport to the syndicate base bathroom when the nuke countdown is initiated

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -418,6 +418,42 @@
 					setDir(i)
 					sleep(1)
 
+/mob/living/simple_animal/pet/dog/corgi/Ian/narsie_act()
+	playsound(src, 'sound/magic/demon_dies.ogg', 75, TRUE)
+	var/mob/living/simple_animal/pet/dog/corgi/narsie/N = new(get_turf(src))
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: use loc

**Review by Cyberboss — DISMISSED** [review]

**Review by Cyberboss — CHANGES_REQUESTED** [review]
^

**Comment by Armhulen** [comment]
I know cobby has a clock cult parrot sprite so the dream is a bit more alive than you think

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 1 above)
Comment (Xhuis) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: In what universe do we prefer `loc` over `get_turf()`?

**Comment by Xhuis** [comment]
If you can get sprites I can do it

**Comment by ExcessiveUseOfCobblestone** [comment]
hahahahhaa wow these?

**Comment by Anonmare** [comment]
_**[BORFS IN ELDRITCH]**_

**Comment by Xhuis** [comment]
If you've got a DMI, cobby, send it to me and I'll make it happen.

**Comment by ExcessiveUseOfCobblestone** [comment]
I've lost the back side but if you are feeling overzealous you could just look it up from the PR #20800

**Comment by Xhuis** [comment]
You're damn right I'm overzealous

**Comment by Poojawa** [comment]
<3 <3 <3

This is my favorite PR since Poly got persistence and his own twitter account.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 2]
```diff
@@ -418,6 +418,42 @@
...
+/mob/living/simple_animal/pet/dog/corgi/narsie/update_corgi_fluff()
+	..()
+	speak = list("Tari'karat-pasnar!", "IA! IA!", "BRRUUURGHGHRHR")
+	speak_emote = list("growls", "barks ominously")
+	emote_hear = list("barks echoingly!", "woofs hauntingly!", "yaps in an eldritch manner.", "mutters something unspeakable.")
+	emote_see = list("communes with the unnameable.", "ponders devouring some souls.", "shakes.")
+
+/mob/living/simple_animal/pet/dog/corgi/narsie/narsie_act()
+	adjustBruteLoss(-maxHealth)
+	return
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Trailing return

File: code/modules/mob/living/simple_animal/parrot.dm [hunk 3]
```diff
@@ -991,3 +998,26 @@
...
+	gold_core_spawnable = FALSE
+
+/mob/living/simple_animal/parrot/clock_hawk/ratvar_act()
+	return
+
+/mob/living/simple_animal/parrot/clock_hawk/death(gibbed)
+	playsound(src, 'sound/magic/clockwork/anima_fragment_death.ogg', 50, TRUE)
+	. = ..()
+	if(!QDELETED(src))
+		qdel(src)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/parrot.dm]: Use del_on_death instead

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 2 above)
Comment (Xhuis) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: It is not; corgis' base `narsie_act()` gibs them and replaces them with a construct

Comment (ChangelingRain) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: It's still a trailing return. You're not calling ..(), so what the parent proc is doesn't matter.

**Comment by JJRcop** [comment]
Nars-Ian and clock hawk should fight like Nar-sie and Ratvar do.
https://github.com/tgstation/tgstation/blob/70ce39d6b3f7065db226d2e5bf78bb35057d8691/code/game/gamemodes/clock_cult/clock_structures/ratvar_the_clockwork_justicar.dm#L109-L150

**Comment by ExcessiveUseOfCobblestone** [comment]
Narsian + Polyvar = NARPOLVARSIAN, The Third Eldrich Lord.

File: code/modules/mob/living/simple_animal/parrot.dm [hunk 4]
```diff
@@ -991,3 +998,26 @@
...
 	H.visible_message("<span class='danger'>[src] dive bombs into [H]'s chest and vanishes!</span>", "<span class='userdanger'>[src] dive bombs into your chest, vanishing! This can't be good!</span>")
+
+
+/mob/living/simple_animal/parrot/clock_hawk
+	name = "clock hawk"
+	desc = "Cbyl jnaan penpxre! Fdhnnnjx!"
+	icon_state = "clock_hawk_fly"
+	icon_living = "clock_hawk_fly"
+	icon_sit = "clock_hawk_sit"
+	speak = list("Penpxre!", "Ratvar vf n qhzo anzr naljnl!")
```
Comment (MoreRobustThanYou) [inline code/modules/mob/living/simple_animal/parrot.dm]: Tvir zr n penpxre, puhpxyrshpx!

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/parrot.dm]: I hope xhuis is doing alright after his horrible keyboard strokes.

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 2 above)
Comment (JJRcop) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: This means you can remove the return and it will work like before.

Comment (Xhuis) [reply code/modules/mob/living/simple_animal/friendly/dog.dm]: I'm aware

File: code/modules/mob/living/simple_animal/parrot.dm [hunk 5]
```diff
@@ -1012,12 +1012,11 @@
 	emote_see = list("flutters its metal wings.")
 	faction = list("ratvar")
 	gold_core_spawnable = FALSE
+	del_on_death = TRUE
 
 /mob/living/simple_animal/parrot/clock_hawk/ratvar_act()
 	return
 
 /mob/living/simple_animal/parrot/clock_hawk/death(gibbed)
 	playsound(src, 'sound/magic/clockwork/anima_fragment_death.ogg', 50, TRUE)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/parrot.dm]: isn't there a var for death sounds

**Pull request merged**